### ISSUE TITLE
[@typescript-eslint/no-explicit-any] Conflicts with mixin class TS2545

### ISSUE THREAD

**Issue by oliversalzburg** [typescript-eslint/typescript-eslint#2408]
- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [read the FAQ](https://github.com/typescript-eslint/typescript-eslint/blob/master/docs/getting-started/linting/FAQ.md) and my problem is not listed.

**Repro**

```JSON
{
  "root": true,
  "env": {
    "node": true,
    "es6": true
  },
  "parser": "@typescript-eslint/parser",
  "plugins": ["@typescript-eslint"],
  "extends": [
    "eslint:recommended",
    "plugin:@typescript-eslint/eslint-recommended",
    "plugin:@typescript-eslint/recommended"
  ],
  "rules": {
    "@typescript-eslint/no-unused-vars": ["off"]
  }
}
```

```TS
type Constructor<T> = { new(...args: any[]): T }

const Mixin = <T extends Constructor<Record<string, unknown>>>(Base: T) =>
	class MixinClass extends Base {
		constructor(...args: any[]) {
			super(...args)
		}
	}
```

**Expected Result**

I want to declare the mixin as clean as possible and there should be a resolution that satisfies ESLint that is also valid TypeScript.

**Actual Result**

ESLint complains about the `any` type in the constructor. When I change it to `unknown`, TypeScript says it's no longer valid code.

> error TS2545: A mixin class must have a constructor with a single rest parameter of type 'any[]'.

Similarly, I don't see how to cleanly declar the `Constructor` type for the mixin.

**Additional Info**

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `3.9.1` |
| `@typescript-eslint/parser`        | `3.9.1` |
| `TypeScript`                       | `3.9.7` |
| `ESLint`                           | `7.7.0` |
| `node`                             | `14.8.0` |

**Comment by bradzacher** [comment]
try the [`ignoreRestArgs`](https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/no-explicit-any.md#ignorerestargs) option.

**Comment by oliversalzburg** [comment]
Yes. That's much better. Thanks.
